enhancements/api-review/gcp_user_defined_labels.md

```diff
@@ -0,0 +1,610 @@
+not create or manage organization resource, hence tags to be applied on the resources
+created by OpenShift must already exist. Tags bindings are inherited by the children of the
+resources. Users can define list of tags to add or to update inherited tags with acceptable values.
+
+Since tags must be pre-existing, OpenShift will not validate the requirements of tag key, tag
+value or tag binding resources and will not define any tags.
+
+Though tags are not created by OpenShift, validations are performed on the user defined tags to
+catch any invalid configurations before requesting GCP service. User defined tags must meet all
+the below conditions or else the cluster creation will fail.
+1. A tag key and value must have minimum of 1 character and can have maximum of 63 characters.
+2. A tag key must contain only uppercase and lowercase alphanumeric characters, hyphens,
+   underscores, and periods.
+3. A tag value must contain only uppercase and lowercase alphanumeric characters, hyphens,
+   underscores, periods, at signs, percent signs, equals signs, plusses, colons, commas,
+   asterisks, pound signs, ampersands, parentheses, square braces, curly braces, and spaces.
+4. A tag key and value must begin and end with an alphanumeric character.
+5. Tag key must already exist.
+6. Tag value must be one of the pre-defined values for the key.
+7. A maximum of 50 tags can be configured.
+8. Tag key with same value shouldn't already exist on a resource(inherited from parent resource)
+
+### Workflow Description
+
+***GCP Labels***
+- An OpenShift administrator requests to add required labels to all GCP resources
+  created by OpenShift by adding it in `platform.gcp.userLabels`
+- OpenShift installer validates the labels defined in `platform.gcp.userLabels` and
+  adds these labels to all resources created during installation and also updates
+  `.status.platformStatus.gcp.resourceLabels` of the `infrastructure.config.openshift.io`
+- In-cluster operators refers `.status.platformStatus.gcp.resourceLabels` of the
+  `infrastructure.config.openshift.io` to add labels to the resources created later.
+
+***GCP Tags***
+- An OpenShift administrator requests to add required tags to all GCP resources
+  created by OpenShift by adding it in `platform.gcp.userTags`
+- OpenShift installer validates the tags defined in `platform.gcp.userTags` and
+  adds these tags to all resources created during installation and also updates
+  `.status.platformStatus.gcp.resourceTags` of the `infrastructure.config.openshift.io`
+- In-cluster operators refers `.status.platformStatus.gcp.resourceTags` of the
+  `infrastructure.config.openshift.io` to add tags to the resources created later.
+
+#### Variation [optional]
+
+### API Extensions
+Enhancement requires below modifications to the mentioned CRDs
+- Add `userLabels` and `userTags` fields to `platform.gcp` of the 
+  `installconfigs.install.openshift.io`
```

Comment:
For persistent volumes this means the labels / tags could not be set by the driver (since this CRD is OpenShift specific), so something else would have to copy the labels from the CRD.